src/install/index.js

```diff
@@ -136,16 +144,18 @@ const preCleanup = () => {
     });
 };
 
+const deleteDeployDoc = (deployDoc) => {
+  debug('Delete deploy ddoc');
+  deployDoc._deleted = true;
+  return DB.app.put(deployDoc);
+};
+
 const postCleanup = (ddocWrapper, deployDoc) => {
   return Promise.all([
         removeOldVersion(ddocWrapper),
         clearStagedDdocs()
       ])
-      .then(() => {
-        debug('Delete deploy ddoc');
-        deployDoc._deleted = true;
-        return DB.app.put(deployDoc);
-      })
+      .then(deleteDeployDoc(deployDoc))
```

Comment:
The syntax here is slightly wrong... I think it should be:

```suggestion
      .then(() => deleteDeployDoc(deployDoc))
```

The difference is when the function is actually going to be executed. Without the wrapping function it executes immediately. To test this theory I wrote this function...

```js
new Promise(resolve => {
  setTimeout(resolve, 100);
})
  .then(() => console.log('one'))
  .then(() => console.log('two'))
  .then(console.log('three'));
```

The output is...

```
> three
> one
> two
```

Reply:
Woah, yes.  Thanks for the explanation!